legend-engine-xt-persistence-test-runner/src/main/java/org/finos/legend/engine/testable/persistence/extension/PersistenceTestRunner.java

```diff
@@ -80,16 +81,33 @@ public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, P
 
         PersistenceTestH2Connection persistenceTestH2Connection = new PersistenceTestH2Connection();
         Connection connection = persistenceTestH2Connection.getConnection();
+        boolean isPersistenceSpecModelV1 = persistence.persister != null ? true : false;
 
         try
         {
-            Dataset targetDataset = DatasetMapper.getTargetDataset(purePersistence);
-            DatasetDefinition datasetDefinition = (DatasetDefinition) targetDataset;
-
             List<AssertionStatus> assertStatuses = new ArrayList<>();
-            Set<String> fieldsToIgnore = IngestModeMapper.getFieldsToIgnore(persistence);
-            boolean isTransactionMilestoningTimeBased = IngestModeMapper.isTransactionMilestoningTimeBased(persistence);
+            Dataset targetDataset;
+            Set<String> fieldsToIgnore;
+            boolean isTransactionMilestoningTimeBased;
+            ServiceOutputTarget serviceOutputTarget = null;
+
+            // Test runner flow for v1
+            if (isPersistenceSpecModelV1)
+            {
+                targetDataset = DatasetMapper.getTargetDataset(purePersistence);
+                fieldsToIgnore = IngestModeMapper.getFieldsToIgnore(persistence);
+                isTransactionMilestoningTimeBased = IngestModeMapper.isTransactionMilestoningTimeBased(persistence);
+            }
+            // Test runner flow for v2
+            else
+            {
+                targetDataset = org.finos.legend.engine.testable.persistence.mapper.v2.DatasetMapper.getTargetDatasetV2(purePersistence);
+                serviceOutputTarget = org.finos.legend.engine.testable.persistence.mapper.v2.IngestModeMapper.getServiceOutputTarget(persistence);
+                fieldsToIgnore = org.finos.legend.engine.testable.persistence.mapper.v2.IngestModeMapper.getFieldsToIgnore(serviceOutputTarget);
+                isTransactionMilestoningTimeBased = org.finos.legend.engine.testable.persistence.mapper.v2.IngestModeMapper.isTransactionMilestoningTimeBased(serviceOutputTarget);
 
+            }
+            DatasetDefinition datasetDefinition = (DatasetDefinition) targetDataset;
             if (!(persistenceTest.isTestDataFromServiceOutput))
```

Comment:
When will this be supported? Having a feature explicitly for testing behavior on the grammar that we explicitly did not develop seems a big gap.  

We should remove the feature or implement to support it.

Reply:
We have a jira in backlog for passing service input data as test data - will check with Dave on the priority and update back

Reply:
Similar as comment above - these inconsistencies hurt the user experience and likely discourage defining test cases.